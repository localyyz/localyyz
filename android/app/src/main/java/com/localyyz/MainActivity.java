package com.localyyz;

import android.content.Intent;
import android.os.Bundle;

import com.facebook.react.ReactActivity;
import com.facebook.react.ReactActivityDelegate;

import javax.annotation.Nullable;

import io.branch.rnbranch.RNBranchModule;

public class MainActivity extends ReactActivity {

    /**
     * Returns the name of the main component registered from JavaScript.
     * This is used to schedule rendering of the component.
     */
    @Override
    protected String getMainComponentName() {
        return "localyyz";
    }

    @Override
    protected void onStart() {
        super.onStart();
        RNBranchModule.initSession(getIntent().getData(), this);
    }

    @Override
    public void onNewIntent(Intent intent) {
        setIntent(intent);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        MainApplication.getCallbackManager().onActivityResult(requestCode, resultCode, data);
    }

    @Override
    protected ReactActivityDelegate createReactActivityDelegate() {
        return new ReactActivityDelegate(this, getMainComponentName()) {
            @Nullable
            @Override
            protected Bundle getLaunchOptions() {
                Bundle initialProps = new Bundle();
                initialProps.putString("API_URL", BuildConfig.API_URL);
                return initialProps;
            }
        };
    }

}
