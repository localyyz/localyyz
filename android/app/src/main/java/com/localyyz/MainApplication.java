package com.localyyz;

import android.app.Application;

import com.BV.LinearGradient.LinearGradientPackage;
import com.cmcewen.blurview.BlurViewPackage;
import com.facebook.CallbackManager;
import com.facebook.FacebookSdk;
import com.facebook.appevents.AppEventsLogger;
import com.facebook.react.ReactApplication;
import com.geektime.rnonesignalandroid.ReactNativeOneSignalPackage;
import com.idehub.GoogleAnalyticsBridge.GoogleAnalyticsBridgePackage;
import com.brentvatne.react.ReactVideoPackage;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.reactnative.androidsdk.FBSDKPackage;
import com.facebook.soloader.SoLoader;
import com.learnium.RNDeviceInfo.RNDeviceInfo;
import com.microsoft.codepush.react.CodePush;
import com.oblador.vectoricons.VectorIconsPackage;
import com.reactnative.photoview.PhotoViewPackage;

import java.util.Arrays;
import java.util.List;

import ca.jaysoo.extradimensions.ExtraDimensionsPackage;
import io.branch.referral.Branch;
import io.branch.rnbranch.RNBranchPackage;

//import com.reactnativepayments.ReactNativePaymentsPackage;

public class MainApplication extends Application implements ReactApplication {

  private static CallbackManager mCallbackManager = CallbackManager.Factory.create();

  protected static CallbackManager getCallbackManager() {
    return mCallbackManager;
  }

  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {

    @Override
    protected String getJSBundleFile() {
      return CodePush.getJSBundleFile();
    }
    
    @Override
    public boolean getUseDeveloperSupport() {
      return BuildConfig.DEBUG;
    }

    @Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
          new MainReactPackage(),
            new ReactNativeOneSignalPackage(),
            new GoogleAnalyticsBridgePackage(),
            new ReactVideoPackage(),
            new ExtraDimensionsPackage(),
              new CodePush(BuildConfig.CODEPUSH_KEY, getApplicationContext(), BuildConfig.DEBUG),
              new RNBranchPackage(),
              //new ReactNativePaymentsPackage(),
              new FBSDKPackage(mCallbackManager),
              new PhotoViewPackage(),
              new VectorIconsPackage(),
              new LinearGradientPackage(),
              new RNDeviceInfo(),
              new BlurViewPackage()
      );
    }

    @Override
    public List<ReactPackage> createAdditionalReactPackages() {
      return getPackages();
    }


    @Override
    protected String getJSMainModuleName() {
      return "index";
    }
  };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

  @Override
  public void onCreate() {
    super.onCreate();

    FacebookSdk.sdkInitialize(getApplicationContext());
    // If you want to use AppEventsLogger to log events.
    AppEventsLogger.activateApp(this);

    // Branch
    Branch.getAutoInstance(this);

    SoLoader.init(this, /* native exopackage */ false);
  }

}
