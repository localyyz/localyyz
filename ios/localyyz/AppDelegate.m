/**
 * Copyright (c) 2015-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "AppDelegate.h"

//#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>

// Branch
#import <react-native-branch/RNBranch.h>

// Facebook
#import <FBSDKCoreKit/FBSDKCoreKit.h>
#import <FBSDKLoginKit/FBSDKLoginKit.h>
#import <FBNotifications/FBNotifications.h>
#import <UserNotifications/UserNotifications.h>

// CodePush
#import <CodePush/CodePush.h>

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  NSURL *jsCodeLocation;

  if (RCT_DEBUG == 1) {
    //jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
    jsCodeLocation = [NSURL URLWithString:@"http://localhost:8081/index.bundle?platform=ios&dev=true"];
  } else {
    jsCodeLocation = [CodePush bundleURL];
  }

  NSString *apiUrl = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"API_URL"];
  NSDictionary *props = @{@"API_URL" : apiUrl};
  
  RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL:jsCodeLocation
                                                      moduleName:@"localyyz"
                                               initialProperties:props
                                                   launchOptions:launchOptions];
  rootView.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];
  
  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];

  // replace RN load view with the launch screen
  UIView* launchScreenView = [[[NSBundle mainBundle] loadNibNamed:@"LaunchScreen" owner:self options:nil] objectAtIndex:0];
  launchScreenView.frame = self.window.bounds;
  rootView.loadingView = launchScreenView;

  // facebook
  [[FBSDKApplicationDelegate sharedInstance] application:application didFinishLaunchingWithOptions:launchOptions];

  // branch
  [RNBranch initSessionWithLaunchOptions:launchOptions isReferrable:YES];

  // push notification
  UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
  center.delegate = self;
  [center requestAuthorizationWithOptions:(UNAuthorizationOptionSound | UNAuthorizationOptionAlert | UNAuthorizationOptionBadge) completionHandler:^(BOOL granted, NSError * _Nullable error) {
    if( !error ) {
      // required to get the app to do anything at all about push notifications
      [[UIApplication sharedApplication] registerForRemoteNotifications];
      NSLog( @"Push registration success.");
    } else {
      NSLog( @"Push registration FAILED" );
      NSLog( @"ERROR: %@ - %@", error.localizedFailureReason, error.localizedDescription );
      NSLog( @"SUGGESTIONS: %@ - %@", error.localizedRecoveryOptions, error.localizedRecoverySuggestion );
    }
  }];

  return YES;
}

- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo fetchCompletionHandler:(nonnull void (^)(UIBackgroundFetchResult))completionHandler {
  FBNotificationsManager *notificationsManager = [FBNotificationsManager sharedManager];
  [notificationsManager presentPushCardForRemoteNotificationPayload:userInfo
                                                 fromViewController:nil
                                                         completion:^(FBNCardViewController * _Nullable viewController, NSError * _Nullable error) {
                                                           if (error) {
                                                             completionHandler(UIBackgroundFetchResultFailed);
                                                           } else {
                                                             completionHandler(UIBackgroundFetchResultNewData);
                                                           }
                                                         }];
}

- (BOOL)application:(UIApplication *)application
            openURL:(NSURL *)url
            options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options {
  
  BOOL handled = [[FBSDKApplicationDelegate sharedInstance] application:application
                                                                openURL:url
                                                      sourceApplication:options[UIApplicationOpenURLOptionsSourceApplicationKey]
                                                             annotation:options[UIApplicationOpenURLOptionsAnnotationKey]
                  ];
  // Add any custom logic here.
  return handled;
}

// track opening and resuming the application attached to the push notification
- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo {
  [FBSDKAppEvents logPushNotificationOpen:userInfo];
}

// track action performed that was attached to the push notification
- (void)application:(UIApplication *)application handleActionWithIdentifier:(NSString *)identifier forRemoteNotification:(NSDictionary *)userInfo
  completionHandler:(void (^)())completionHandler {
  [FBSDKAppEvents logPushNotificationOpen:userInfo action:identifier];
}

- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray *restorableObjects))restorationHandler {
  return [RNBranch continueUserActivity:userActivity];
}

- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
  NSString *token = [[deviceToken description] stringByTrimmingCharactersInSet: [NSCharacterSet characterSetWithCharactersInString:@"<>"]];
  token = [token stringByReplacingOccurrencesOfString:@" " withString:@""];
  NSLog(@"content--- %@", token);
  [FBSDKAppEvents setPushNotificationsDeviceToken:deviceToken];
}

// Facebook SDK
- (void)applicationDidBecomeActive:(UIApplication *)application {
  [FBSDKAppEvents activateApp];
}

@end
