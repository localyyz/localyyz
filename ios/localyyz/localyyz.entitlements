<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>aps-environment</key>
	<string>development</string>
	<key>com.apple.developer.associated-domains</key>
	<array>
		<string>applinks:localyyz.app.link</string>
		<string>applinks:localyyz-alternate.app.link</string>
		<string>applinks:localyyz.test-app.link</string>
		<string>applinks:localyyz-alternate.test-app.link</string>
	</array>
	<key>com.apple.developer.in-app-payments</key>
	<array>
		<string>merchant.com.localyyz.toronto</string>
	</array>
</dict>
</plist>
