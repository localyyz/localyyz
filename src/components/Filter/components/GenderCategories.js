import React from "react";
import { ActivityIndicator, StyleSheet, View, FlatList } from "react-native";

// third party
import { inject, observer } from "mobx-react/native";
import Accordion from "react-native-collapsible/Accordion";

// custom
import { Sizes, Styles } from "localyyz/constants";

// local
import Category from "./Category";
import Gender from "./Gender";
import SelectedFilter from "./SelectedFilter";

// constants
const GENDER_SECTION = [
  { label: "Women", value: "woman" },
  { label: "Men", value: "man" }
];

@inject(stores => ({
  fetchCategories: stores.filterStore.fetchCategories,
  clearFilter: stores.filterStore.clearCategoryFilter,
  categories: stores.filterStore.categories || [],
  selected: stores.filterStore.subcategory || stores.filterStore.category,
  gender: stores.filterStore.gender
}))
@observer
export default class Categories extends React.Component {
  constructor(props) {
    super(props);
    this.state = { activeSection: 0 };

    // bindings
    this.renderItem = this.renderItem.bind(this);
    this._renderHeader = this._renderHeader.bind(this);
    this._renderContent = this._renderContent.bind(this);
  }

  componentDidMount() {
    this.props.fetchCategories();
  }

  componentDidUpdate(prevProps) {
    if (prevProps.gender !== this.props.gender) {
      this.setState({ activeSection: this.props.gender == "woman" ? 0 : 1 });
    }
  }

  renderItem({ item: category }) {
    return <Category {...category} />;
  }

  _renderHeader(content /*index, isActive, sections*/) {
    return (
      <View style={styles.row}>
        <Gender label={content.label} value={content.value} />
      </View>
    );
  }

  // TODO: this should really respect the title and not just use the value name
  // currently not possible since category lists are gone when drilling down

  _renderContent(/*content, index, isActive, sections*/) {
    return (
      <View style={styles.content}>
        {this.props.selected ? (
          <View style={styles.selected}>
            <SelectedFilter onClear={this.props.clearFilter}>
              {this.props.selected}
            </SelectedFilter>
          </View>
        ) : null}
        <FlatList
          scrollEnabled={false}
          data={this.props.categories.slice()}
          keyExtractor={item => item.title}
          renderItem={this.renderItem}
          ListFooterComponent={
            <ActivityIndicator
              style={styles.footer}
              size="small"
              hidesWhenStopped={true}
              animating={!this.props.categories.length}/>
          }/>
      </View>
    );
  }

  render() {
    return (
      <View style={styles.container}>
        <Accordion
          activeSection={this.state.activeSection}
          sections={GENDER_SECTION}
          renderHeader={this._renderHeader}
          renderContent={this._renderContent}/>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {},

  content: {
    alignItems: "flex-start"
  },

  row: {
    ...Styles.Horizontal,
    ...Styles.EqualColumns
  },

  selected: {
    marginVertical: Sizes.InnerFrame
  },

  footer: {
    paddingVertical: Sizes.InnerFrame / 3
  }
});
