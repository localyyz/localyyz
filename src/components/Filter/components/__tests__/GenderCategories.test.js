import React from "react";

// custom
import { findByKey, findByRef, renderWithLayout } from "localyyz/tests";
import ShallowRenderer from "react-test-renderer/shallow";

// local
import GenderCategories, { GENDER_SECTION } from "../GenderCategories";
let Component = GenderCategories.wrappedComponent;
let shallowRenderer = new ShallowRenderer();

// constants
const NAME = "Category filter";
const PROPS = {
  categories: [
    {
      title: "Apparel",
      value: "apparel"
    },
    {
      title: "Shoes",
      value: "shoes"
    }
  ],
  fetchCategories: jest.fn()
};

describe(NAME, () => {
  it("should render properly", () => {
    shallowRenderer.render(<Component {...PROPS} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toBeDefined();

    const content = findByRef(result, "genderCategories");
    expect(content).toBeDefined();
    expect(content.props.sections).toEqual(
      expect.arrayContaining(GENDER_SECTION)
    );
    expect(content).toMatchSnapshot();
  });

  it("should render content properly", () => {
    shallowRenderer.render(<Component {...PROPS} />);
    const result = shallowRenderer
      .getRenderOutput()
      .props.children.props.renderContent();

    expect(findByRef(result, "genderCategoriesContent")).toBeDefined();
    const list = findByRef(result, "genderCategoriesList");
    expect(list).toBeDefined();
    expect(list.props.data).toEqual(expect.arrayContaining(PROPS.categories));

    expect(result).toMatchSnapshot();
    expect(list).toMatchSnapshot();
  });

  it("should render content with selected properly", () => {
    shallowRenderer.render(<Component {...PROPS} selected="apparel" />);
    const content = shallowRenderer
      .getRenderOutput()
      .props.children.props.renderContent();

    expect(content.props.children[0]).toBeTruthy();

    expect(content).toMatchSnapshot();
  });

  it("should render header properly", () => {
    shallowRenderer.render(<Component {...PROPS} />);
    const result = shallowRenderer
      .getRenderOutput()
      .props.children.props.renderHeader(GENDER_SECTION[0]);
    expect(result).toBeDefined();

    const header = findByKey(result, GENDER_SECTION[0].value);
    expect(header).toBeDefined();
    expect(header.props.value).toEqual(GENDER_SECTION[0].value);
    expect(header.props.label).toEqual(GENDER_SECTION[0].label);

    expect(header).toMatchSnapshot();
  });

  it("should render items properly", () => {
    shallowRenderer.render(<Component {...PROPS} />);
    const result = shallowRenderer
      .getRenderOutput()
      .props.children.props.renderContent();

    const list = findByRef(result, "genderCategoriesList");
    const item = list.props.renderItem({ item: PROPS.categories[0] });
    expect(item.key).toEqual(PROPS.categories[0].value);
    expect(item.props).toEqual(PROPS.categories[0]);

    expect(item).toMatchSnapshot();
  });

  it("should render to snapshot", () => {
    expect(
      renderWithLayout(<Component {...PROPS} />, null, true)
    ).toMatchSnapshot();
  });
});
