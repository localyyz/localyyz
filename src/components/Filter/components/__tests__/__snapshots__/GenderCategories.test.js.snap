// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category filter should render content properly 1`] = `
<Component
  style={
    Object {
      "alignItems": "flex-start",
    }
  }
>
  <FlatList
    ListFooterComponent={
      <ActivityIndicator
        animating={false}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
        style={
          Object {
            "paddingVertical": 5,
          }
        }
      />
    }
    data={
      Array [
        Object {
          "title": "Apparel",
          "value": "apparel",
        },
        Object {
          "title": "Shoes",
          "value": "shoes",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</Component>
`;

exports[`Category filter should render content properly 2`] = `
<FlatList
  ListFooterComponent={
    <ActivityIndicator
      animating={false}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
      style={
        Object {
          "paddingVertical": 5,
        }
      }
    />
  }
  data={
    Array [
      Object {
        "title": "Apparel",
        "value": "apparel",
      },
      Object {
        "title": "Shoes",
        "value": "shoes",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEnabled={false}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`Category filter should render content with selected properly 1`] = `
<Component
  style={
    Object {
      "alignItems": "flex-start",
    }
  }
>
  <Component
    style={
      Object {
        "marginVertical": 15,
      }
    }
  >
    <SelectedFilter>
      apparel
    </SelectedFilter>
  </Component>
  <FlatList
    ListFooterComponent={
      <ActivityIndicator
        animating={false}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
        style={
          Object {
            "paddingVertical": 5,
          }
        }
      />
    }
    data={
      Array [
        Object {
          "title": "Apparel",
          "value": "apparel",
        },
        Object {
          "title": "Shoes",
          "value": "shoes",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</Component>
`;

exports[`Category filter should render header properly 1`] = `
<inject-Gender
  label="Women"
  value="woman"
/>
`;

exports[`Category filter should render items properly 1`] = `
<withNavigation(inject-Category)
  title="Apparel"
  value="apparel"
/>
`;

exports[`Category filter should render properly 1`] = `
<Accordion
  activeSection={0}
  disabled={false}
  renderContent={[Function]}
  renderHeader={[Function]}
  sections={
    Array [
      Object {
        "label": "Women",
        "value": "woman",
      },
      Object {
        "label": "Men",
        "value": "man",
      },
    ]
  }
  touchableComponent={[Function]}
  underlayColor="black"
/>
`;

exports[`Category filter should render to snapshot 1`] = `
<Component
  style={Object {}}
>
  <Accordion
    activeSection={0}
    disabled={false}
    renderContent={[Function]}
    renderHeader={[Function]}
    sections={
      Array [
        Object {
          "label": "Women",
          "value": "woman",
        },
        Object {
          "label": "Men",
          "value": "man",
        },
      ]
    }
    touchableComponent={[Function]}
    underlayColor="black"
  />
</Component>
`;
